src/ngx_http_lua_upstream_module.c

```diff
@@ -349,6 +353,55 @@ ngx_http_lua_upstream_set_peer_down(lua_State * L)
 }
 
 
+static int
+ngx_http_lua_upstream_get_chosen_peer(lua_State * L)
+{
+    ngx_str_t                           host;
+    ngx_http_upstream_srv_conf_t       *us;
+    ngx_http_request_t                 *r;
+    ngx_http_request_t                 *fake_r;
+    ngx_peer_connection_t              *peer;
+    int                                 rc;
+
+    if (lua_gettop(L) != 1) {
+        return luaL_error(L, "exactly one argument expected");
+    }
+
+    host.data = (u_char *) luaL_checklstring(L, 1, &host.len);
+
+    us = ngx_http_lua_upstream_find_upstream(L, &host);
+    if (us == NULL) {
+        lua_pushnil(L);
+        lua_pushliteral(L, "upstream not found");
+        return 2;
+    }
+
+    r = ngx_http_lua_get_request(L);
+
+    // we create a fake request object, which we will use to fetch the upstream
```

Comment:
Do you want me to remove the comments completely, or just change them to `/* */` ?